app/routes/synonyms.$synonymId.tsx

```diff
@@ -44,12 +56,12 @@ export async function action({
     const subtractions =
       getFormDataString(data, 'deleteSynonyms')?.split(',') || ([] as string[])
     const filtered_subtractions = subtractions.filter((sub) => sub)
-    await putSynonyms({
+    const response = await putSynonyms({
       synonymId,
       additions: filtered_additions,
       subtractions: filtered_subtractions,
     })
-    return null
+    return response?.error
```

Comment:
Generally the front-end code should be written to assume that the code takes the happy path. Use an [error boundary](https://remix.run/docs/en/main/route/error-boundary) if you want to fine-tune how errors are presented.

```suggestion
    return null
```